eppo_core/src/eval/eval_details.rs

```diff
@@ -0,0 +1,245 @@
+    Match,
+    /// Configuration has not been fetched yet.
+    ConfigurationMissing,
+    /// Flag does not exist or is not enabled for the environment in use.
+    FlagUnrecognizedOrDisabled,
+    /// Default allocation is matched and is also serving NULL, resulting in the default value being
+    /// assigned.
+    DefaultAllocationNull,
+    /// Variation value does not match the specified type for the function called.
+    TypeMismatch,
+    /// Configuration received from the server is invalid for the SDK. This should normally never
+    /// happen and is likely a signal that you should update SDK.
+    UnexpectedConfigurationError,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BanditEvaluationCode {
+    /// Found a bandit action.
+    Match,
+    /// Configuration has not been fetched yet.
+    ConfigurationMissing,
+    /// Configuration received from the server is invalid for the SDK. This should normally never
+    /// happen and is likely a signal that you should update SDK.
+    UnexpectedConfigurationError,
+    /// Assignment evaluated to a non-bandit variation.
+    NonBanditVariation,
+    /// `get_bandit_action` was called without supplying actions.
+    NoActionsSuppliedForBandit,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvaluationResultWithDetails<T> {
+    pub variation: Option<T>,
+    pub action: Option<String>,
+    pub evaluation_details: EvaluationDetails,
+}
+
+impl<T> EvaluationResultWithDetails<T> {
+    /// Map `EvaluationResultWithDetails.variation` using the `f` function.
+    pub fn map<T2, F: FnOnce(T) -> T2>(self, f: F) -> EvaluationResultWithDetails<T2> {
+        EvaluationResultWithDetails {
+            variation: self.variation.map(f),
+            action: self.action,
+            evaluation_details: self.evaluation_details,
+        }
+    }
+}
+
+/// Details about feature flag or bandit evaluation.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvaluationDetails {
+    pub flag_key: String,
+    pub subject_key: String,
+    pub subject_attributes: Attributes,
+    /// Timestamp when the flag was evaluated.
+    pub timestamp: DateTime<Utc>,
+
+    /// Timestamp when configuration was fetched by the SDK. None if configuration hasn't been
+    /// fetched yet.
+    pub config_fetched_at: Option<DateTime<Utc>>,
+    /// Timestamp when configuration was published by the server. None if configuration hasn't been
+    /// fetched yet.
+    pub config_published_at: Option<DateTime<Utc>>,
+    /// Environment the configuration belongs to. None if configuration hasn't been fetched yet.
+    pub environment_name: Option<String>,
+
+    pub bandit_evaluation_code: Option<BanditEvaluationCode>,
+    pub flag_evaluation_code: Option<FlagEvaluationCode>,
```

Comment:
One new deviation from JS SDK: `bandit_evaluation_code` and `flag_evaluation_code` are exposed separately. The reasoning is that overriding `flag_evaluation_code` with bandit evaluation result may hide some interesting details (i.e., why the given variation was selected)